stdlib/messaging/nats/src/main/java/org/ballerinalang/nats/basic/producer/CloseConnection.java

```diff
@@ -41,7 +41,8 @@ public static void closeConnection(ObjectValue producerObject) {
         if (TypeChecker.getType(connection).getTag() == TypeTags.OBJECT_TYPE_TAG) {
             ObjectValue connectionObject = (ObjectValue) connection;
             ((AtomicInteger) connectionObject.getNativeData(Constants.CONNECTED_CLIENTS)).decrementAndGet();
-            NatsMetricsUtil.reportProducerClose(connectionObject.getStringValue(Constants.URL));
+            String[] urls = connectionObject.getArrayValue(Constants.URL).getStringArray();
+            NatsMetricsUtil.reportProducerClose(String.join(",", urls));
```

Comment:
Can we use a fine-grained method to construct a comma-separated URL from the connection object as this has similar occurrences?

```ballerina
    private static String getCommaSeparatedUrl(ObjectValue connectionObject) {
        return String.join(",", connectionObject.getArrayValue(Constants.URL).getStringArray());
    }
```

Reply:
Fixed.